pkg/bgpv1/agent/controller.go

```diff
@@ -187,10 +200,22 @@ func (c *Controller) Run(ctx context.Context) {
 			return
 		case <-c.Sig.Sig:
 			l.Info("Cilium BGP Control Plane Controller woken for reconciliation")
+			if retrying {
+				if err := bo.Wait(ctx); err != nil {
```

Comment:
Should ongoing retry be cancelled if there is new change which triggers reconciliation ?

Main issue I see here is bo.Wait can block for long in between retries. This can cause delay in applying new changes.

Reply:
Good point. We currently don't have any way to distinguish the event triggered by retry and the event triggered by configuration change. One way to solve this is putting event origin information into `sig.Event()`. So that we can bypass retry when the event origin is a configuration change.

However, I think it's better, in general, to apply the rate limit here even if the event origin is a configuration change. This is a separate topic, though, because retry and rate limit are different things.

Reply:
Hmm, this is a though one to decide, but maybe we could apply the same approach as k8s controller-runtime applies, so that we behave similarly to standard k8s controllers. IIRC, in controller-runtime only retries are rate-limited, new changes are reconciled immediately.

See:
 - `c.Queue.AddRateLimited(req)` on error [here](https://github.com/kubernetes-sigs/controller-runtime/blob/b88ed7a3602b85b6cfd0acfe9c25033b978bdb83/pkg/internal/controller/controller.go#L315-L329)
 - `q.Add` (wirthout rate-limiting) on CRUD events [here](https://github.com/kubernetes-sigs/controller-runtime/blob/b88ed7a3602b85b6cfd0acfe9c25033b978bdb83/pkg/handler/enqueue.go#L46-L50)

Reply:
As we agreed in the offline conversation, we'll reduce the scope of this PR and discuss separately about retry. Let me close this discussion.